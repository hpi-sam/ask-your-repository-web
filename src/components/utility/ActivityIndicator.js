// @flow
import React from 'react';
import classNames from 'classnames';
import { MdCropSquare } from 'react-icons/md';
import './ActivityIndicator.scss';

type Props = {
  className?: string,
  text?: string,
};

function ActivityIndicator({ text, className }: Props) {
  const styleClasses = classNames('ActivityIndicator', className);

  return (
    <div className={styleClasses}>
      <MdCropSquare className="ActivityIndicator__spinner" />
      {text && (
        <div className="ActivityIndicator__text">
          {text}
        </div>
      )}
    </div>
  );
}

ActivityIndicator.defaultProps = {
  className: '',
  text: '',
};

export default ActivityIndicator;
