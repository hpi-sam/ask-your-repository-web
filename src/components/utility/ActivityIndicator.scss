.ActivityIndicator {
  @keyframes spin {
    from { transform: rotate(0deg); }
    to { transform: rotate(360deg); }
  }

  @keyframes color-pulse {
    0% { color: var(--main-text-color); }
    50% { color: rgb(36, 186, 255); }
    100% { color: var(--main-text-color); }
  }

  display: flex;
  flex-direction: column;
  align-items: center;
  color: var(--main-text-color);

  &__spinner {
    font-size: 3rem;
    animation:
      spin 2s infinite,
      color-pulse 4s infinite;
  }

  &__text {
    margin-top: 20px;
    font-size: 1.2rem;
    font-weight: bold;
    animation: color-pulse 4s infinite;
  }
}
