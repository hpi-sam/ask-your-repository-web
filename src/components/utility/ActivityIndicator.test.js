// @flow
import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';
import ActivityIndicator from './ActivityIndicator';

describe('<ActivityIndicator />', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<ActivityIndicator text="Loading" />);
  });

  it('renders correctly', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
