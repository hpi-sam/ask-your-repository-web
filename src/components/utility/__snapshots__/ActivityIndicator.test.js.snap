// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ActivityIndicator /> renders correctly 1`] = `
<div
  className="ActivityIndicator"
>
  <MdCropSquare
    className="ActivityIndicator__spinner"
  />
  <div
    className="ActivityIndicator__text"
  >
    Loading
  </div>
</div>
`;
